installer/pkg/common/ca.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package common
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	cacerts      = "cacerts"
+	etcSSLCerts  = "/etc/ssl/certs"
+	caVolumeName = "gitpod-ca-certificate"
+	caCrtKey     = "ca.crt"
+
+	updateContainerdScript = `TARGETS="docker containerd";for TARGET in $TARGETS;do mkdir -p /etc/$TARGET/certs.d/reg.%s:%v && echo "$GITPOD_CA_CERT" > /etc/$TARGET/certs.d/reg.%s:%v/ca.crt && echo "OK";done`
+)
+
+func InternalCAVolumeMount() *corev1.VolumeMount {
+	return &corev1.VolumeMount{
+		Name:      cacerts,
+		ReadOnly:  true,
+		MountPath: etcSSLCerts,
+	}
+}
+
+func InternalCAVolume() *corev1.Volume {
+	return &corev1.Volume{
+		Name: caVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		},
+	}
+}
+
+// InternalCAContainer returns a container that updates the CA certificates
+// adding the internal Gitpod self-generated CA.
+// This is required for components that communicate with registry-facade
+// and cannot use certificates signed by unknown authority, like containerd or buildkit
+func InternalCAContainer() *corev1.Container {
+	return &corev1.Container{
+		Name: "update-ca-certificates",
+		// TODO: move this image to gitpod.io
+		Image: ImageName("ghcr.io/aledbf/", "gitpod-ca-updater", "latest"),
+		Args: []string{
+			"bash",
+			"-c",
+			"set -e; update-ca-certificates -f; cp /etc/ssl/certs/* /ssl-certs; echo 'OK'",
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      cacerts,
+				MountPath: "/ssl-certs",
+			},
+			{
+				Name:      caVolumeName,
+				SubPath:   caCrtKey,
+				MountPath: "/usr/local/share/ca-certificates/gitpod-ca.crt",
+			},
+		},
+	}
+}
+
+func UpdateContainerdCertificate(ctx *RenderContext, containerPort int) *corev1.Container {
+	return &corev1.Container{
+		Name:    "update-containerd-certificates",
+		Image:   ImageName("docker.io", "alpine", "3.14"),
+		Command: []string{"nsenter"},
+		Args: []string{
+			"--mount=/proc/1/ns/mnt", "--", "sh", "-c", "$(SETUP_SCRIPT)",
```

Comment:
Could we instead of container escaping (hence the `hostPID`, no?) just mount the target path?

Reply:
done